<!DOCTYPE html>
<html lang='en'>
<head>
<title>libressl-2.7.patch\python3\main - aports - Main aports tree
</title>
<meta name='generator' content='cgit v1.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='//alpinelinux.org/alpine-logo.ico'/>
<link rel='alternate' title='Atom feed' href='https://git.alpinelinux.org/cgit/aports/atom/main/python3/libressl-2.7.patch?h=master' type='application/atom+xml'/>
<link rel='vcs-git' href='git://git.alpinelinux.org/aports' title='aports Git repository'/>
<link rel='vcs-git' href='https://git.alpinelinux.org/cgit/aports' title='aports Git repository'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/cgit/'><img src='//wiki.alpinelinux.org/w/resources/assets/alogo.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/cgit/'>index</a> : <a title='aports' href='/cgit/aports/'>aports</a></td><td class='form'><form method='get'>
<select name='h' onchange='this.form.submit();'>
<option value='1.10-stable'>1.10-stable</option>
<option value='1.9'>1.9</option>
<option value='2.0-stable'>2.0-stable</option>
<option value='2.1-stable'>2.1-stable</option>
<option value='2.2-stable'>2.2-stable</option>
<option value='2.3-stable'>2.3-stable</option>
<option value='2.4-stable'>2.4-stable</option>
<option value='2.5-stable'>2.5-stable</option>
<option value='2.6-stable'>2.6-stable</option>
<option value='2.7-stable'>2.7-stable</option>
<option value='3.0-stable'>3.0-stable</option>
<option value='3.1-stable'>3.1-stable</option>
<option value='3.2-stable'>3.2-stable</option>
<option value='3.3-stable'>3.3-stable</option>
<option value='3.4-stable'>3.4-stable</option>
<option value='3.5-stable'>3.5-stable</option>
<option value='3.6-stable'>3.6-stable</option>
<option value='3.7-stable'>3.7-stable</option>
<option value='master' selected='selected'>master</option>
</select> <input type='submit' value='switch'/></form></td></tr>
<tr><td class='sub'>Main aports tree
</td><td class='sub right'>gitolite</td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/aports/about/'>about</a><a href='/cgit/aports/'>summary</a><a href='/cgit/aports/refs/'>refs</a><a href='/cgit/aports/log/main/python3/libressl-2.7.patch'>log</a><a class='active' href='/cgit/aports/tree/main/python3/libressl-2.7.patch'>tree</a><a href='/cgit/aports/commit/main/python3/libressl-2.7.patch'>commit</a><a href='/cgit/aports/diff/main/python3/libressl-2.7.patch'>diff</a><a href='/cgit/aports/stats/main/python3/libressl-2.7.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/aports/log/main/python3/libressl-2.7.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/aports/tree/'>root</a>/<a href='/cgit/aports/tree/main'>main</a>/<a href='/cgit/aports/tree/main/python3'>python3</a>/<a href='/cgit/aports/tree/main/python3/libressl-2.7.patch'>libressl-2.7.patch</a></div><div class='content'>blob: 6ecb473bde58436089dcf33f1363a4b2451f1f2e (<a href='/cgit/aports/plain/main/python3/libressl-2.7.patch'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
<a id='n89' href='#n89'>89</a>
<a id='n90' href='#n90'>90</a>
<a id='n91' href='#n91'>91</a>
<a id='n92' href='#n92'>92</a>
<a id='n93' href='#n93'>93</a>
<a id='n94' href='#n94'>94</a>
<a id='n95' href='#n95'>95</a>
<a id='n96' href='#n96'>96</a>
<a id='n97' href='#n97'>97</a>
<a id='n98' href='#n98'>98</a>
<a id='n99' href='#n99'>99</a>
<a id='n100' href='#n100'>100</a>
<a id='n101' href='#n101'>101</a>
<a id='n102' href='#n102'>102</a>
<a id='n103' href='#n103'>103</a>
<a id='n104' href='#n104'>104</a>
<a id='n105' href='#n105'>105</a>
<a id='n106' href='#n106'>106</a>
<a id='n107' href='#n107'>107</a>
<a id='n108' href='#n108'>108</a>
<a id='n109' href='#n109'>109</a>
<a id='n110' href='#n110'>110</a>
<a id='n111' href='#n111'>111</a>
</pre></td>
<td class='lines'><pre><code>From f5befbb0d1526f18eb2b24eabb48c3b761c624a2 Mon Sep 17 00:00:00 2001
From: Christian Heimes &lt;christian&#64;python.org&gt;
Date: Sat, 24 Mar 2018 18:38:14 +0100
Subject: [PATCH] [3.6] bpo-33127: Compatibility patch for LibreSSL 2.7.0
 (GH-6210) (GH-6214)

LibreSSL 2.7 introduced OpenSSL 1.1.0 API. The ssl module now detects
LibreSSL 2.7 and only provides API shims for OpenSSL &lt; 1.1.0 and
LibreSSL &lt; 2.7.

Documentation updates and fixes for failing tests will be provided in
another patch set.

Signed-off-by: Christian Heimes &lt;christian&#64;python.org&gt;.
(cherry picked from commit 4ca0739c9d97ac7cd45499e0d31be68dc659d0e1)

Co-authored-by: Christian Heimes &lt;christian&#64;python.org&gt;
<span style="color:#0057ae">---</span>
 Lib/test/test_ssl.py                               |  1 +
 .../2018-03-24-15-08-24.bpo-33127.olJmHv.rst       |  1 +
 Modules/_ssl.c                                     | 24 ++++++++++++++--------
 Tools/ssl/multissltests.py                         |  3 ++-
 4 files changed, 20 insertions(+), 9 deletions(-)
 create mode 100644 Misc/NEWS.d/next/Library/2018-03-24-15-08-24.bpo-33127.olJmHv.rst

diff --git a/Lib/test/test_ssl.py b/Lib/test/test_ssl.py
index 8dd3b4145078..9785a59a7e49 100644
<span style="color:#0057ae">--- a/Lib/test/test_ssl.py</span>
<span style="color:#000000; font-weight:bold">+++ b/Lib/test/test_ssl.py</span>
&#64;&#64; -1687,6 +1687,7 &#64;&#64; def test_get_ca_certs_capath(self):
         self.assertEqual(len(ctx.get_ca_certs()), 1)
 
     &#64;needs_sni
<span style="color:#000000; font-weight:bold">+    &#64;unittest.skipUnless(hasattr(ssl, &quot;PROTOCOL_TLSv1_2&quot;), &quot;needs TLS 1.2&quot;)</span>
     def test_context_setget(self):
         # Check that the context of a connected socket can be replaced.
         ctx1 = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
diff --git a/Misc/NEWS.d/next/Library/2018-03-24-15-08-24.bpo-33127.olJmHv.rst b/Misc/NEWS.d/next/Library/2018-03-24-15-08-24.bpo-33127.olJmHv.rst
new file mode 100644
index 000000000000..635aabbde031
<span style="color:#0057ae">--- /dev/null</span>
<span style="color:#000000; font-weight:bold">+++ b/Misc/NEWS.d/next/Library/2018-03-24-15-08-24.bpo-33127.olJmHv.rst</span>
<span style="color:#010181">&#64;&#64; -0,0 +1 &#64;&#64;</span>
<span style="color:#000000; font-weight:bold">+The ssl module now compiles with LibreSSL 2.7.1.</span>
diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index c54e43c2b48a..5e007da858bd 100644
<span style="color:#0057ae">--- a/Modules/_ssl.c</span>
<span style="color:#000000; font-weight:bold">+++ b/Modules/_ssl.c</span>
&#64;&#64; -106,6 +106,12 &#64;&#64; struct py_ssl_library_code {
 
 #if (OPENSSL_VERSION_NUMBER &gt;= 0x10100000L) &amp;&amp; !defined(LIBRESSL_VERSION_NUMBER)
 #  define OPENSSL_VERSION_1_1 1
<span style="color:#000000; font-weight:bold">+#  define PY_OPENSSL_1_1_API 1</span>
<span style="color:#000000; font-weight:bold">+#endif</span>
<span style="color:#000000; font-weight:bold">+</span>
<span style="color:#000000; font-weight:bold">+/* LibreSSL 2.7.0 provides necessary OpenSSL 1.1.0 APIs */</span>
<span style="color:#000000; font-weight:bold">+#if defined(LIBRESSL_VERSION_NUMBER) &amp;&amp; LIBRESSL_VERSION_NUMBER &gt;= 0x2070000fL</span>
<span style="color:#000000; font-weight:bold">+#  define PY_OPENSSL_1_1_API 1</span>
 #endif
 
 /* Openssl comes with TLSv1.1 and TLSv1.2 between 1.0.0h and 1.0.1
&#64;&#64; -152,16 +158,18 &#64;&#64; struct py_ssl_library_code {
 #define INVALID_SOCKET (-1)
 #endif
 
<span style="color:#0057ae">-#ifdef OPENSSL_VERSION_1_1</span>
<span style="color:#0057ae">-/* OpenSSL 1.1.0+ */</span>
<span style="color:#0057ae">-#ifndef OPENSSL_NO_SSL2</span>
<span style="color:#0057ae">-#define OPENSSL_NO_SSL2</span>
<span style="color:#0057ae">-#endif</span>
<span style="color:#0057ae">-#else /* OpenSSL &lt; 1.1.0 */</span>
<span style="color:#0057ae">-#if defined(WITH_THREAD)</span>
<span style="color:#000000; font-weight:bold">+/* OpenSSL 1.0.2 and LibreSSL needs extra code for locking */</span>
<span style="color:#000000; font-weight:bold">+#if !defined(OPENSSL_VERSION_1_1) &amp;&amp; defined(WITH_THREAD)</span>
 #define HAVE_OPENSSL_CRYPTO_LOCK
 #endif
 
<span style="color:#000000; font-weight:bold">+#if defined(OPENSSL_VERSION_1_1) &amp;&amp; !defined(OPENSSL_NO_SSL2)</span>
<span style="color:#000000; font-weight:bold">+#define OPENSSL_NO_SSL2</span>
<span style="color:#000000; font-weight:bold">+#endif</span>
<span style="color:#000000; font-weight:bold">+</span>
<span style="color:#000000; font-weight:bold">+#ifndef PY_OPENSSL_1_1_API</span>
<span style="color:#000000; font-weight:bold">+/* OpenSSL 1.1 API shims for OpenSSL &lt; 1.1.0 and LibreSSL &lt; 2.7.0 */</span>
<span style="color:#000000; font-weight:bold">+</span>
 #define TLS_method SSLv23_method
 #define TLS_client_method SSLv23_client_method
 #define TLS_server_method SSLv23_server_method
&#64;&#64; -227,7 +235,7 &#64;&#64; SSL_SESSION_get_ticket_lifetime_hint(const SSL_SESSION *s)
     return s-&gt;tlsext_tick_lifetime_hint;
 }
 
<span style="color:#0057ae">-#endif /* OpenSSL &lt; 1.1.0 or LibreSSL */</span>
<span style="color:#000000; font-weight:bold">+#endif /* OpenSSL &lt; 1.1.0 or LibreSSL &lt; 2.7.0 */</span>
 
 
 enum py_ssl_error {
diff --git a/Tools/ssl/multissltests.py b/Tools/ssl/multissltests.py
index ce5bbd85308c..ba4529ae0611 100755
<span style="color:#0057ae">--- a/Tools/ssl/multissltests.py</span>
<span style="color:#000000; font-weight:bold">+++ b/Tools/ssl/multissltests.py</span>
<span style="color:#010181">&#64;&#64; -57,8 +57,9 &#64;&#64;</span>
 ]
 
 LIBRESSL_RECENT_VERSIONS = [
<span style="color:#0057ae">-    &quot;2.5.3&quot;,</span>
     &quot;2.5.5&quot;,
<span style="color:#000000; font-weight:bold">+    &quot;2.6.4&quot;,</span>
<span style="color:#000000; font-weight:bold">+    &quot;2.7.1&quot;,</span>
 ]
 
 # store files in ../multissl
</code></pre></td></tr></table>
</div> <!-- class=content -->
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-2804446-12");
pageTracker._trackPageview();
} catch(err) {}</script>
</div> <!-- id=cgit -->
</body>
</html>
